### PR TITLE
Update java version command with config file

### PR DESCRIPTION
**Pull request by stefanTolksdorf**
## What

Update java version command with config file

### CODE REVIEW

File: pontos/version/commands/_java.py [hunk 1]
```diff
@@ -15,199 +15,160 @@
...
             changed_files=changed_files,
         )
+
+    def _update_version_files(self, new_version) -> List[Path]:
+        config = self._load_config()
+
+        changed_files: List[Path] = []
+        for file_config in config["files"]:
+            file_path = file_config["path"]
+            with open(Path.cwd() / file_path, "r") as input_file_handle:
```
Comment (bjoernricks) [inline pontos/version/commands/_java.py:88]:
```suggestion
            with (Path.cwd() / file_path).open("r") as input_file_handle:
```

But as far as I can see you just need the text of the file therefore you can just use
```
content = (Path.cwd() / file_path).read_text(encoding="utf8")
```

File: pontos/version/commands/_java.py [hunk 2]
```diff
@@ -15,199 +15,160 @@
...
+
+    def _load_config(self) -> Dict:
+        version_config_file = Path.cwd() / "upgradeVersion.json"
+        if not version_config_file.exists():
+            raise VersionError(
+                f"No {version_config_file} config file found. "
+                "This file is required for pontos"
+            )
+
+        with open(version_config_file, "r") as f:
```
Comment (bjoernricks) [inline pontos/version/commands/_java.py:121]:
```suggestion
        with version_config_file.open("r") as f:
```

File: pontos/version/commands/_java.py [hunk 3]
```diff
@@ -15,199 +15,160 @@
...
+                    )
+                lines[line_number - 1] = (
+                    matches.group("pre")
+                    + str(new_version)
+                    + matches.group("post")
+                )
+
+                content = "".join(lines)
+                with open(Path.cwd() / file_path, "w") as output_file_handle:
+                    output_file_handle.write(content)
```
Comment (bjoernricks) [inline pontos/version/commands/_java.py:109]:
```suggestion
               (Path.cwd() / file_path).write_text(content, encoding="utf8")
```

File: pontos/version/commands/_java.py [hunk 4]
```diff
@@ -15,199 +15,160 @@
...
+        config = self._load_config()
+
+        file_versions = {}
+        for file_config in config["files"]:
+            file_path = file_config["path"]
+            file = Path.cwd() / file_path
+            if not file.exists():
+                raise VersionError(f"No {file} file found.")
+
+            with open(file, "r") as f:
```
Comment (bjoernricks) [inline pontos/version/commands/_java.py:136]:
```suggestion
            with file.open("r") as f:
```

File: pontos/version/commands/_java.py [hunk 5]
```diff
@@ -15,199 +15,160 @@
...
+                    + matches.group("post")
+                )
+
+                content = "".join(lines)
+                with open(Path.cwd() / file_path, "w") as output_file_handle:
+                    output_file_handle.write(content)
+                changed_files.append(Path(file_config["path"]))
+        return changed_files
+
+    def _load_config(self) -> Dict:
```
Comment (bjoernricks) [inline pontos/version/commands/_java.py:113]:
```suggestion
    def _load_config(self) -> Dict[str, Any]:
```
Maybe the dict can be typed even more precise like `Dict[str, Union[int, str]]`?

File: pontos/version/commands/_java.py [hunk 6]
```diff
@@ -15,199 +15,160 @@
...
+                    raise VersionError(
+                        f"Line has no version, "
+                        f"file:'{file_path}' "
+                        f"lineNo:{line_number} "
+                        f"content:'{version_line}'"
+                    )
+                file_versions[file_path] = matches.group("version")
+        return file_versions
+
+    def _verify_version(self, file_versions: Dict) -> str:
```
Comment (bjoernricks) [inline pontos/version/commands/_java.py:157]:
```suggestion
    def _verify_version(self, file_versions: Dict[str, str]) -> str:
```

File: pontos/version/commands/_java.py [hunk 7]
```diff
@@ -15,199 +15,160 @@
...
+                f"No {version_config_file} config file found. "
+                "This file is required for pontos"
+            )
+
+        with open(version_config_file, "r") as f:
+            json_string = f.read()
+            config = json.loads(json_string)
+            return config
+
+    def _read_versions_from_files(self) -> Dict:
```
Comment (bjoernricks) [inline pontos/version/commands/_java.py:126]:
```suggestion
    def _read_versions_from_files(self) -> Dict[str, str]:
```

File: pontos/version/commands/_java.py [hunk 8]
```diff
@@ -15,199 +15,160 @@
...
             changed_files=changed_files,
         )
+
+    def _update_version_files(self, new_version) -> List[Path]:
+        config = self._load_config()
+
+        changed_files: List[Path] = []
+        for file_config in config["files"]:
+            file_path = file_config["path"]
+            with (Path.cwd() / file_path).open("r") as input_file_handle:
```
Comment (y0urself) [inline pontos/version/commands/_java.py:88]:
```suggestion
            with (Path.cwd() / file_path).open("r", encoding="utf-8") as input_file_handle:
```

File: pontos/version/commands/_java.py [hunk 9]
```diff
@@ -15,199 +15,160 @@
...
+                        f"content:'{version_line}'"
+                    )
+                lines[line_number - 1] = (
+                    matches.group("pre")
+                    + str(new_version)
+                    + matches.group("post")
+                )
+
+                content = "".join(lines)
+                with (Path.cwd() / file_path).open("w") as output_file_handle:
```
Comment (y0urself) [inline pontos/version/commands/_java.py:108]:
```suggestion
                with (Path.cwd() / file_path).open("w", encoding="utf-8") as output_file_handle:
```

Comment (stefanTolksdorf) [reply pontos/version/commands/_java.py:108]: not all files are UTF-8

**Review by bjoernricks — APPROVED** [review]

**Pull request merged**